### PR TITLE
Run travis in SLES containers

### PR DESCRIPTION
**Pull request by ChrisBr**
We can now build SLES images in OBS so we use these images in Travis to test against SLES.

The main benefits are
- we have the same setup & base system as we use in production
- no need to maintain Ubuntu for Travis

Supersedes https://github.com/openSUSE/open-build-service/pull/4677

### CODE REVIEW

File: contrib/start_lint [hunk 1]
```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+ls -la /obs/src/api
```
Comment (eduardoj) [inline contrib/start_lint]: :thought_balloon: What is this needed for? Maybe a leftover?

Comment (ChrisBr) [reply contrib/start_lint]: debugging, it is still failing

**Comment by adrianschroeter** [comment]
we are already in the middle of upgrading to SLE 15, so you may want to switch over already ...

**Comment by ChrisBr** [comment]
> we are already in the middle of upgrading to SLE 15, so you may want to switch over already ...

As long as we run build.o.o on SLE 12 I would stick to SLE12

File: contrib/start_minitest [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 echo "Preparing application..."
-bundle exec rake dev:bootstrap[old_test_suite] RAILS_ENV=test
+bundle exec rake dev:bootstrap[old_test_suite] RAILS_ENV=test &> /dev/null
```
Comment (hennevogel) [inline contrib/start_minitest]: Sounds like a bad idea. What if this goes wrong?

File: dist/ci/travis_script.sh [hunk 3]
```diff
@@ -4,41 +4,26 @@
...
-      perl -pi -e 's/source_port: 5352/source_port: 3200/' config/options.yml
-      bundle exec rails test:api
+      rake docker:test:minitest
       ;;
     spider)
-      unset DO_COVERAGE
-      bundle exec rails assets:precompile &> /dev/null
-      perl -pi -e 's/source_port: 5352/source_port: 3200/' config/options.yml
-      bundle exec rails test:spider
+      DO_COVERAGE=1
```
Comment (hennevogel) [inline dist/ci/travis_script.sh]: We want coverage from the spider test? No...

File: contrib/start_minitest (hunk 2 above)
Comment (ChrisBr) [reply contrib/start_minitest]: Agreed, removed!

**Comment by ChrisBr** [comment]
Hooray, everything passes now :smile: 
So this one is now ready for review :eyes:

**Comment by hennevogel** [comment]
## Latest without docker
screenshot from 2018-04-16 13-24-20

## Latest with docker
screenshot from 2018-04-16 13-24-55

**Comment by hennevogel** [comment]
@ChrisBr is using the old test suite outside CI still possible? Like https://github.com/openSUSE/open-build-service/wiki/Development-Environment#run-a-single-test-file-in-the-old-api-test-suite

**Comment by hennevogel** [comment]
Step                   | docker (s) | ubuntu (s)
-------------------- | ------------- | ---------
git checkout       | 2.12          | 2.25
git submodule    | 0.88          | 0.89
services             | 0.02          | 0.03
rvm                    | 22.83        | 21.45
before scripts     | 0.03          | 27.72 (18.85 + 0.42 + 8.45)
suite                   | 1246.24    | 1007.19

File: contrib/start_lint [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+echo "Installing bundle..."
+export NOKOGIRI_USE_SYSTEM_LIBRARIES=1; bundle install --jobs=3 --retry=3 || bundle install --jobs=3 --retry=3
+sudo npm install -g jshint
```
[the remark this answers is not in the document]
Comment (ChrisBr) [reply contrib/start_lint]: We could install the package
https://build.opensuse.org/package/show/devel:languages:nodejs/jshint

However, that would mean that we do use the latest version always?

[the remark this answers is not in the document]
Comment (ChrisBr) [reply contrib/start_lint]: I would postpone this to a later PR

File: .gitignore [hunk 5]
```diff
@@ -70,3 +70,4 @@ nbproject
 *.iws
 .idea/
 .kanku/
+src/api/.bundle/
```
[the remark this answers is not in the document]
Comment (ChrisBr) [reply .gitignore]: Done

**Comment by ChrisBr** [comment]
With the bundle cache we can get it a little bit faster but it is still a little bit slower than running it directly in the Travis machine.

The big change for the minitest testsuite also might come that we run now more tests in Travis. Before we skipped certain tests if the system != (open)SUSE (e.g. https://github.com/openSUSE/open-build-service/blob/master/src/api/test/functional/published_controller_test.rb#L159).

@openSUSE/open-build-service are you fine with these changes?

File: Rakefile [hunk 6]
```diff
@@ -57,6 +57,16 @@ namespace :docker do
         sh 'docker-compose -f docker-compose.ci.yml -p minitest stop'
       end
     end
+
+    desc 'Run the spider test to crawl all pages and fail for exceptions'
```
Comment (bgeuken) [inline Rakefile]: and fail in case of exceptions

File: Rakefile [hunk 7]
```diff
@@ -57,6 +57,16 @@ namespace :docker do
...
+    task :spider do
+      begin
+        cmd = '../../contrib/start_spider'
+        sh "docker-compose -f docker-compose.ci.yml run --rm minitest /bin/bash -c '#{cmd}'"
```
Comment (bgeuken) [inline Rakefile]: I think it's better to break the line here. It's not like we need this variable anywhere else.

File: .travis.yml [hunk 8]
```diff
@@ -33,9 +33,8 @@ jobs:
     - stage: Linters
```
Comment (bgeuken) [inline .travis.yml]: This commit looks like it belongs to one of the previous ones

Comment (ChrisBr) [reply .travis.yml]: Can you please tell me which commits you want to get squashed?

Comment (bgeuken) [reply .travis.yml]: This one: https://github.com/openSUSE/open-build-service/pull/4853/commits/e54545894df03bb1cc93d60a35cbc6e0ae5e6629

**Pull request merged**